cellxgene_schema_cli/cellxgene_schema/atac_seq.py

```diff
@@ -0,0 +1,355 @@
+        # the python gzip library.
+        unzipped_file = Path(tempdir) / Path(fragment_file).name.replace(".gz", "")
+        logger.info(f"Unzipping {fragment_file}")
+        with open(unzipped_file, "wb") as fp:
+            subprocess.run(["gunzip", "-c", fragment_file], stdout=fp, check=True)
+
+        _dask_cluster_config = dict(silence_logs=logging.ERROR, dashboard_address=None)
+        if dask_cluster_config:
+            _dask_cluster_config.update(dask_cluster_config)
+
+        with dd.LocalCluster(**_dask_cluster_config) as cluster, dd.Client(cluster):
+            # convert the fragment to a parquet file
+            logger.info(f"Converting {fragment_file} to parquet")
+            parquet_file_path = Path(tempdir) / Path(fragment_file).name.replace(".gz", ".parquet")
+            ddf.read_csv(unzipped_file, sep="\t", names=column_ordering, dtype=column_types).to_parquet(
+                parquet_file_path, partition_on=["chromosome"], compute=True
+            )
+
+            # remove the unzipped file
+            logger.debug(f"Removing {unzipped_file}")
+            unzipped_file.unlink()
+            parquet_file = str(parquet_file_path)
+            errors = validate(parquet_file, anndata_file)
+            if any(errors):
+                logger.error("Errors found in Fragment and/or Anndata file")
+                logger.error(errors)
+                return False
+            else:
+                logger.info("Fragment and Anndata file are valid")
+
+            if generate_index:
+                logger.info(f"Sorting fragment and generating index for {fragment_file}")
+                index_fragment(fragment_file, parquet_file, tempdir)
+        logger.debug("cleaning up")
+
+
+def validate(parquet_file: str, anndata_file: str) -> list[Optional[str]]:
+    jobs = [
+        validate_fragment_start_coordinate_greater_than_0(parquet_file),
+        validate_fragment_barcode_in_adata_index(parquet_file, anndata_file),
+        validate_fragment_stop_coordinate_within_chromosome(parquet_file, anndata_file),
+        validate_fragment_stop_greater_than_start_coordinate(parquet_file),
+        validate_fragment_read_support(parquet_file),
+        validate_anndata_is_atac(anndata_file),
+        validate_anndata_feature_reference(anndata_file),
+        validate_anndata_is_primary_data(anndata_file),
+    ]
+    return jobs
+
+
+def validate_fragment_start_coordinate_greater_than_0(parquet_file: str) -> Optional[str]:
+    df = ddf.read_parquet(parquet_file, columns=["start coordinate"])
+    series = df["start coordinate"] > 0
+    if not series.all().compute():
+        return "Start coordinate must be greater than 0."
+
+
+def validate_fragment_barcode_in_adata_index(parquet_file: str, anndata_file: str) -> Optional[str]:
+    df = ddf.read_parquet(parquet_file, columns=["barcode"])
+    obs = ad.read_h5ad(anndata_file, backed="r").obs
```

Comment:
```suggestion
    with h5py.File(anndata_file) as f:
        obs = ad.io.read_elem(f["obs"])
```

^ This will read less, and could be significantly faster depending on how much other data is in the file.

If you are on anndata < 0.11 `read_elem` is under `anndata.experimental`